core/ppu.cpp

```diff
@@ -88,29 +100,103 @@ void PPU::step(u8 cpuCyclesElapsed) {
   switch(mode) {
     case OAM:
       if (cyclesLeft >= OAM_CLOCKS) {
-
+        cyclesLeft -= OAM_CLOCKS;
+        
+        // update stat register for VRAM mode
+        u8 stat = get_stat();
+        stat = setBit(stat, 0);
+        stat = setBit(stat, 1);
+        mmu.write(STAT, stat);
+        mode = VRAM;
       }
       break;
     case VRAM:
-      if (cyclesLeft >= VRAM_CLCOKS) {
+      if (cyclesLeft >= VRAM_CLOCKS) {
+        cyclesLeft -= VRAM_CLOCKS;
+
+        drawScanLine();
+        
+        mode = HBLANK;
+        u8 stat = get_stat();
+        stat = clearBit(stat, 0);
+        stat = clearBit(stat, 1);
+        mmu.write(STAT, stat);
 
+        // check for HBLANK interrupt
+        if (checkBit(get_stat(), 3)) {
+          // signal cpu interrupt?
+        }
       }
       break;
     case HBLANK:
       if (cyclesLeft >= HBLANK_CLOCKS) {
+        cyclesLeft -= HBLANK_CLOCKS;
+
+        u8 scanline = get_ly();
+
+        // check lyc interupt // should this be somewhere else?
+        bool lyc = (scanline == get_lyc());
+        if (checkBit(get_stat(), 6) && lyc) {
+          // signal cpu interrupt?
+        }
+        
+        // increment current line
+        scanline++;
+        mmu.write(LY, scanline);
 
+        // check current scanline == 144, then enter VBLANK, else enter OAM to prepare to draw another line
+        if (scanline == 144) {
```

Comment:
I have a (likely irrational) fear of skipping over numbers and missing equality checks like this. It probably doesn't matter, but `>=` would make me feel better.